Class Crawler.py

```diff
@@ -0,0 +1,168 @@
+import requests
+import re
+from bs4 import BeautifulSoup
+import random
+import datetime
+random.seed(datetime.datetime.now())
+
+    
+class Crawler(object):
+    def __init__(self, address):
+        
+        self.address = address
+        self.html = requests.get(address)
+        self.soup = BeautifulSoup(self.html.content, "lxml")
+        
+    
+    def getAddress(self):
+        
+        return self.address
+        
+    def domainName(self):
+        '''
+        return domain name of given website
+        '''
+        address = self.address
+        if "https://" in address:
+            address = address.replace("https://", "")
+        if "http://" in address:
+            address = address.replace("http://", "")
+        if "www." in address:
+            address = address.replace("www.", "")
+        domain = address.split('.')[0]
+        return domain
+
+    def getLinks(self):
+        '''
+        returns list of links (strings) that appears on given website
+        '''
+        links_on_site = []
+        soup = self.soup
+        for link in  soup.findAll('a', attrs={'href': re.compile("^(http|www)")}):
+            links_on_site.append(link.attrs['href'])            
+#           print(link.attrs['href'])               # function working check  
+#           print(domainName(link.attrs['href']))   # "domainName" function working check
+        return links_on_site
+    
+    def getRandomLink(self):
```

Comment:
It's better to merge functionality of functions `getRandomLink` and `getRandomLink_extern` into one function and as a input provide list from which it should select randomly element. In that way it will be more generic and less code may be written.

Reply:
Something simple like that: 
`    def getRandomLink(self, links_list):
        '''
        ranomly picks up one extern of the links from the site en returns it
        '''        
        return links_list[random.randint(0, len(links_list)-1)]`